data-products/src/main/scala/org/sunbird/analytics/job/report/StateAdminReportJob.scala

```diff
@@ -114,6 +118,37 @@ object StateAdminReportJob extends optional.Application with IJob with StateAdmi
         saveUserSelfDeclaredExternalInfo(userExternalDecryptData, finalUserDf)
     }
     
+    def appendUserProfileTypeWithLocation(userDf: DataFrame)(implicit sparkSession: SparkSession): DataFrame = {
+        import sparkSession.implicits._
+        val userProfileDf = userDf.select(col("profileusertype"), col("profilelocation"), col("userid")).rdd.map{f =>
+            var usertype: String = "";
+            var usersubtype: String = "";
+            if (null != f.getString(0) && f.getString(0).nonEmpty) {
+                val profileUserType = JSONUtils.deserialize[Map[String, String]](f.getString(0))
+                val usertypeList = profileUserType.filter(f => f._1.equalsIgnoreCase("type")).map(f => f._2)
+                usertype = if (!usertypeList.isEmpty) usertypeList.head else ""
+                val usersubtypeList = profileUserType.filter(f => f._1.equalsIgnoreCase("subType")).map(f => f._2)
+                usersubtype = if (!usersubtypeList.isEmpty) usersubtypeList.head else ""
+            }
+            var locations = new ListBuffer[String]()
+            if (null != f.getString(1) && f.getString(1).nonEmpty) {
+                val loc = JSONUtils.deserialize[List[Map[String, String]]](f.getString(1))
+                val stateIdList = loc.filter(f => f.getOrElse("type", "").equalsIgnoreCase("state")).map(f => f.getOrElse("id", ""))
```

Comment:
Use UDF functions on dataframe

Reply:
@SanthoshVasabhaktula : done the changes please check.